software/contrib/euclid.py

```diff
@@ -62,6 +64,8 @@ def generate_euclidean_pattern(steps, pulses, rot=0):
         raise ValueError
     if steps == 0:
         return []
+    if pulses == 0:
+        return [0]*steps
```

Comment:
The commit message for that change got lost in a bad rebase.  This change fixes a bug where if you tried generating a euclidean rhythm with N steps and zero pulses the function crashed with a division-by-zero error.  So instead if `pulses` is zero just return `[0, 0, ...., 0]` indicating a euclidean rhythm of length N where all steps are off.

Reply:
Thanks for the explanation. In the future it would be ideal to have but fixes for separate scripts in their own PR. Aside from reducing the general scope of this PR, the bug fix would likely have been able to be merged already.

Reply:
The bug only became apparent when implementing Pam's euclidean generation, which is why I just fixed it in this branch.  The Euclidean script itself works in its current state.

Depending how long this PR takes I can cherry-pick that fix into a separate merge request though.

Reply:
Thanks for the explanation. It now makes sense to me for this change to be in this PR. I don't think you need to change anything.